File: java/com/epicdragonworld/gameserver/network/packets/receivable/LocationUpdate.java
package com.epicdragonworld.gameserver.network.packets.receivable;

import com.epicdragonworld.gameserver.managers.WorldManager;
import com.epicdragonworld.gameserver.model.Location;
import com.epicdragonworld.gameserver.model.actor.Player;
import com.epicdragonworld.gameserver.network.GameClient;
import com.epicdragonworld.gameserver.network.ReceivablePacket;
import com.epicdragonworld.gameserver.network.packets.sendable.MoveToLocation;

/**
 * @author Pantelis Andrianakis
 */
public class LocationUpdate
{
	public LocationUpdate(GameClient client, ReceivablePacket packet)
	{
		// Read data.
		final float posX = (float) packet.readDouble(); // TODO: Client WriteFloat
		final float posY = (float) packet.readDouble(); // TODO: Client WriteFloat
		final float posZ = (float) packet.readDouble(); // TODO: Client WriteFloat
		final float heading = (float) packet.readDouble(); // TODO: Client WriteFloat
		final int animState = packet.readShort();
		final int waterState = packet.readByte();
		
		// Update player location.
		final Player player = client.getActiveChar();
		final Location location = player.getLocation();
		location.setX(posX);
		location.setY(posY);
		location.setZ(posZ);
		location.setHeading(heading);
		
		// Broadcast movement.
		for (Player nearby : WorldManager.getVisiblePlayers(player))
		{
			nearby.channelSend(new MoveToLocation(player, heading, animState, waterState));
		}
	}
}

File: EpicDragonWorldServer/Network/Packets/Receivable/LocationUpdate.cs
/**
 * @author Pantelis Andrianakis
 */
public class LocationUpdate
{
    public LocationUpdate(GameClient client, ReceivablePacket packet)
    {
        // Read data.
        float posX = (float)packet.ReadDouble(); // TODO: Client WriteFloat
        float posY = (float)packet.ReadDouble(); // TODO: Client WriteFloat
        float posZ = (float)packet.ReadDouble(); // TODO: Client WriteFloat
        float heading = (float)packet.ReadDouble(); // TODO: Client WriteFloat
        int animState = packet.ReadShort();
        int waterState = packet.ReadByte();

        // Update player location.
        Player player = client.GetActiveChar();
        LocationHolder location = player.GetLocation();
        location.SetX(posX);
        location.SetY(posY);
        location.SetZ(posZ);
        location.SetHeading(heading);

        // Broadcast movement.
        foreach (Player nearby in WorldManager.GetVisiblePlayers(player))
        {
            nearby.ChannelSend(new MoveToLocation(player, heading, animState, waterState));
        }
    }
}
